modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/client/ItThinClientNullsTest.java

```diff
@@ -0,0 +1,136 @@
+
+        expectNotSupported(() -> view.getNullable(null, 1));
+        expectNotSupported(() -> view.getNullableAndPut(null, 1, null));
+        expectNotSupported(() -> view.getNullableAndRemove(null, 1));
+        expectNotSupported(() -> view.getNullableAndReplaceAsync(null, 1, null));
+    }
+
+    @Test
+    public void testModifyOpPoJoNullVal() {
+        Table table = ignite().tables().table("T0");
+        KeyValueView<Integer, BoolStr> view = table.keyValueView(Mapper.of(Integer.class), Mapper.of(BoolStr.class));
+
+        expectNullNotAllowed(() -> view.put(null, 2, null));
+        expectNullNotAllowed(() -> view.putIfAbsent(null, 2, null));
+        expectNullNotAllowed(() -> view.replace(null, 2, null));
+        expectNullNotAllowed(() -> view.replace(null, 2, new BoolStr(), null));
+        expectNullNotAllowed(() -> view.getAndPut(null, 2, null));
+        expectNullNotAllowed(() -> view.getAndReplace(null, 2, null));
+        expectNullNotAllowed(() -> {
+            Map<Integer, BoolStr> map = new HashMap<>();
+            map.put(2, null);
+            view.putAll(null, map);
+        });
+    }
+
+    @Test
+    public void testNullableOpSingleCol() {
+        Table table = ignite().tables().table("T1");
+        KeyValueView<Integer, Boolean> view = table.keyValueView(Mapper.of(Integer.class), Mapper.of(Boolean.class));
+
+        assertEquals(NullableValue.of(null), view.getNullable(null, 1));
+        assertEquals(NullableValue.of(null), view.getNullableAndPut(null, 1, null));
+        assertEquals(NullableValue.of(null), view.getNullableAndRemove(null, 1));
+        assertNull(view.getNullableAndReplace(null, 1, null));
+    }
+
+    @Test
+    public void testModifyOpSingleColNullVal() {
+        Table table = ignite().tables().table("T1");
+        KeyValueView<Integer, Boolean> view = table.keyValueView(Mapper.of(Integer.class), Mapper.of(Boolean.class));
+
+        assertTrue(view.replace(null, 1, null, false));
+        assertEquals(false, view.get(null, 1));
+
+        assertTrue(view.remove(null, 1, false));
+
+        assertTrue(view.putIfAbsent(null, 1, null));
+        view.put(null, 1, true);
+
+        assertEquals(true, view.getAndPut(null, 1, null));
+        assertNull(view.getAndReplace(null, 2, null));
+
+        Map<Integer, Boolean> map = new HashMap<>();
+        map.put(3, null);
+        view.putAll(null, map);
+        assertEquals(NullableValue.of(null), view.getNullable(null, 3));
+    }
+
+    private static void expectNotSupported(Executable exec) {
+        UnsupportedOperationException err = assertThrows(UnsupportedOperationException.class, exec);
+        assertEquals("`getNullable`* methods cannot be used when a value is not mapped to a simple type", err.getMessage());
+    }
+
+    private static void expectNullNotAllowed(Executable exec) {
+        NullPointerException err = assertThrows(NullPointerException.class, exec);
```

Comment:
we have `IgniteTestUtils.assert*` methods

Reply:
fixed